tests/test_chromatic.py

```diff
@@ -1201,6 +1225,269 @@ def test_centroid():
+    # versions is very fast, so it won't slow down the tests too much to do both ways.  Note that
+    # for initial tests (fair comparison between exact and interpolated version) we're going to use
+    # a lot of images for the interpolation, n_interp=100.  For more slowly-evaluated profiles, this
+    # is not going to be feasible, and tests need to be done to check that a given n_interp is
+    # enough.  But that's an accuracy test, not a unit test.
+    sigma_0 = 0.06
+    scale = 0.02
+    n_interp = 100
+    oversample_fac = 1.5
+    blue_limit = min(bandpass.blue_limit, bandpass_g.blue_limit)
+    red_limit = max(bandpass.red_limit, bandpass_g.red_limit)
+    waves = np.linspace(blue_limit, red_limit, n_interp)
+
+    # First, compare images that are drawn with exact and interpolated PSF.  For the latter, we
+    # should check that when we force exact evaluation of the profile at each wavelength, that the
+    # results are also correct.
+    exact_psf = ChromaticGaussian(sigma_0, None)
+    interp_psf = ChromaticGaussian(sigma_0, waves, oversample_fac=oversample_fac)
+    im_exact = exact_psf.drawImage(bandpass, scale=scale)
+    im_interp = im_exact.copy()
+    im_interp = interp_psf.drawImage(bandpass, image=im_interp, scale=scale)
+    im_force = im_exact.copy()
+    im_force = interp_psf.drawImage(bandpass, image=im_force, scale=scale, force_eval=True)
+    # Note: peak value of array is around 1, so going to 4 decimal places is a reasonably stringent
+    # test considering how different the exact vs. interpolated rendering process is.
+    np.testing.assert_array_almost_equal(
+        im_interp.array, im_exact.array, decimal=4,
+        err_msg='InterpolatedChromaticObject results differ for exact vs. interpolated')
+    np.testing.assert_array_almost_equal(
+        im_force.array, im_exact.array, decimal=4,
+        err_msg='InterpolatedChromaticObject results differ for exact vs. forced')
+
+    # Just to be safe, we're going to do our own brute-force calculation of the interpolated result,
+    # and compare it with `im_interp`.  Note that if the definition of the ChromaticGaussian gets
+    # changed, then this code has to get changed too.
+    im = im_exact.copy()
+    im.setZero()
+    for wave in waves:
+        obj = galsim.Gaussian(sigma=sigma_0*(wave/500.), flux=bandpass(wave))
+        obj = obj.shear(g1=0.1*((wave/500.)-1.))
+        obj.drawImage(image=im, scale=scale, add_to_image=True)
+    # Flux normalization should not be the same, but the rest should be.  So just renormalize.
+    im *= im_interp.array.sum()/im.array.sum()
+    np.testing.assert_array_almost_equal(
+        im_interp.array, im.array, decimal=4,
+        err_msg='InterpolatedChromaticObject results differ for interpolated vs. brute force')
+
+    # Also check that with a tiny / narrow band, we get exactly what we expect.
+    # Since the band is extremely narrow we can do this test to quite high accuracy.
+    wave_list = [550., 555., 560.]
```

Comment:
Since exact_psf doesn't have a wave_list, this object only evaluates at precisely these 3 wavelengths, getting multiplied by zero for 2 of them.  I.e., using `wave_list = [1., 555., 560.]` gives exactly the same results.  I'm not sure if that's what you intended...